### ISSUE TITLE
Oferecer o programa em um instalador

### ISSUE THREAD

**Issue by DaviMedrade** [davidsantos-br/ETS2Sync-Helper-2#6]
Se o programa for oferecido em um instalador em vez de (ou além de) uma versão “portable”, ao menos duas vantagens serão obtidas:
1. Se torna possível adicionar uma exceção no firewall para ele, visto que o executável do Ruby estará sempre no mesmo lugar.
2. O programa abrirá mais rápido porque não precisará se auto-desempacotar numa pasta temporária cada vez que for aberto.

A gem utilizada para gerar o executável (`ocra`) já oferece suporte a criar um instalador usando Inno Setup, então isso facilitaria as coisas. Infelizmente, isso aumentaria a complexidade do processo de auto-atualização (#2), mas é bem provável que isso possa ser contornado de forma relativamente simples.

@JhowDroid

**Comment by JhowDroid** [comment]
Opa, quando se tem o pacote instalado, arquivos de configuração .conf podem ser criados para guardar preferências do usuário como idioma, último modo aberto: ATS/ETS #4.
Pense que o processo de instalação fica visível só no momento da instalação, na atualização pode passar despercebido. E conseguir exceção no firewall é um ponto muito positivo, quando se tem usuários de Avast com problemas de falso positivo.
Pense também na possibilidade de criar um serviço de atualização assim como a aplicação flash player e java RE criam.

**Comment by j1m1l0k0** [comment]
@davidsantos-br davidsantos-br, acredito que nem tenha necessidade de regras de firewall.  
@JhowDroid, Já em relação ao anti-virus não existem relação falso-positivo diretamente com o firewall e sim com a assinatura do executável em questão. Como é todo feito em Ruby não acredito que o AV reclame de alguma forma.

**Comment by DaviMedrade** [comment]
Na verdade mesmo com um portable é possível criar um arquivo de configurações desde que seja numa pasta pré-determinada (e.g. Documentos ou AppData). E, se tudo o mais falhar, sempre tem o Registro. :)

**Comment by DaviMedrade** [comment]
Ups. Cliquei sem querer.

**Comment by DaviMedrade** [comment]
@trev0rbr, alguns usuários relataram que o AVG detecta o próprio Ruby como vírus (sabe-se lá por que…).

**Comment by j1m1l0k0** [comment]
@davidsantos-br porque AVG não presta rss

**Comment by DaviMedrade** [comment]
Pois é, eu até pensei em sugerir no README que o usuário mude de anti-vírus se usa o AVG, mas não acho que seria bem-recebido… :stuck_out_tongue:

**Comment by j1m1l0k0** [comment]
@davidsantos-br acho mais fácil deixar um aviso que quem usar AVG corre o risco de ter problemas ao usar o seu programa.    infelizmenteo AVG tem uma base de assinaturas totalmente desconexa.

**Comment by DaviMedrade** [comment]
Na verdade, isso foi bem lembrado. Quando postei sobre o programa no fórum do TruckersMP e no vídeo tutorial eu mencionei essa questão do AVG, mas não cheguei a colocar no README.

**Comment by j1m1l0k0** [comment]
@davidsantos-br alias muitos não sabem mais usar antivirus e jogar ao mesmo tempo não combina e não da certo, perde-se muita performance de GPU e CPU quando o AV está ativo no sistema. Claro que isso não tem nada haver com o foco do assunto mais eu asseguro 100% que não rola.  Principalmente se for Norton ahah   Avast, AVG, Norton, Mcafee e todos os outros.  o AV quando ativo fica escaneando os threads abertos na memória pq ele entende como codigo malicioso e ai que vem o problema de retardamento de recurso.  a CPU fica 100% + GPU e se um jogo deveria rodar 30 FPS vai rodar 12 FPS ou até menos.  (que fique como dica)  rss

**Comment by JhowDroid** [comment]
Bem, tem antivirus que tem o "modo jogo" que entende quando voce está em tela cheia e com maior uso de memória e processamento e para de ficar fazendo scaneamento, mantendo apenas a verificação de tráfego de rede no caso dos que tem firewall.
Mas de qualquer forma, adicionar a pasta da aplicação na exceção do antivirus resolve isso. Mas acredito que seja necessário ter um diretório padrão de instalação para isso.

**Comment by DaviMedrade** [comment]
No caso do AVG, adicionar o executável, mesmo portable, aparentemente resolve (de acordo com o teste rápido que fiz no PC de um amigo que usa AVG).

**Comment by JhowDroid** [comment]
mesmo ele sendo executado em uma pasta temporária após extração automática? já que a pasta temporária não foi especificada na exceção.

**Comment by DaviMedrade** [comment]
Sim. Isso também me pegou de surpresa. Eu tava acessando a máquina do meu amigo por TeamViewer e decidi fazer esse teste só pra desencargo de consciência e calha que funcionou.

**Comment by DaviMedrade** [comment]
Talvez o AVG detecta que os arquivos vieram de um executável na lista de exceções e extende a exceção pra eles, ou talvez ele só detecta como vírus o Ruby dentro de um executável empacotado pelo Ocra.

**Comment by JhowDroid** [comment]
isso significaria que o AVG é burro ou inteligente? kkk

**Comment by DaviMedrade** [comment]
Num dia que eu estiver com mais tempo eu instalo o AVG numa VM só pra ver exatamente em quais situações ele dá esse falso-positivo. Se for só no executável empacotado, uma versão instalada também resolveria isso (a menos que ele também detecte como vírus o Ruby empacotado em um instalador).

**Comment by JhowDroid** [comment]
Sendo mais prático, acentuando aqui a necessidade de uma versão instalada, de uma hora pra outra outro antivirus pode achar o app malicioso conforme forem entrando novas funcionalidades nele. Algumas ações da aplicação podem passar a ser suspeitas, pois não só a assinatura de virus é usada, mas o scan de atividades suspeitas também.

**Comment by j1m1l0k0** [comment]
@davidsantos-br @JhowDroid o problema que as assinaturas de virus são milhões dentro da base AV
(obviamente o AVG e alguns outros não tomam o devido cuidado de filtrar essa base)
o que acontece é que quando AV detecta o endereço de assinatura do arquivo e compara com a base dele isso pode ser repetido com algum vírus já existente por aí.  Outra possibilidade que ao fazer scaneamento no pacote dos arquivos, ele entende como um processo desconhecido e o recurso Eurístico do AV fica tentando encontrar uma solução para o mesmo e quando isso não ocorre ele aponta o arquivo como vírus.
